docs/DEVELOPING.md

```diff
@@ -229,6 +231,19 @@ a panic here is acceptable.
 
 All potential panics *MUST* be clearly documented in the function documentation.
 
+#### Const strings
+
+When re-typing raw string literals repeatedly, this can lead to typo errors,
+especially when names ares simiilar. In general, when reasonable, it is preferred
+to use [Compile-time constants](https://doc.rust-lang.org/std/keyword.const.html)
+when dealing with non-dynamic strings. For example, when working with field names
+for event metadata.
+
+This has not always been a consistently enforced code style for the project, so
+despite seeing usages of raw strings, please take the opportunity when writing
+new code or modifying existing, to use constants or update existing raw strings
+to use constants.
```

Comment:
Could we clarify this to say "strings used more than once?"

Reply:
I also realized that maybe this makes more sense in the style guide? https://github.com/vectordotdev/vector/blob/master/STYLE.md